### ISSUE TITLE
error at lastal step

### ISSUE THREAD

**Issue by DrPintoThe2nd** [dib-lab/dammit#104]
Hi Camille

Thanks for your work on this software! 

Here is my issue:

I reinstalled dammit [v 1.0.rc0] (to an HPC w/ ubuntu v16.04, 48 thread, 128 Gb RAM) via instructions provided here: https://angus.readthedocs.io/en/2017/dammit_annotation.html and have run into this error:

(py3) bjp:~$ dammit annotate transcripts.fasta --busco-group tetrapoda --n_threads 24
# dammit
## a tool for easy de novo transcriptome annotation

by Camille Scott

**v1.0.rc0**, 2018

#### Info
* Database Directory: /home/bjp/.dammit/databases
* Doit Database: /home/bjp/.dammit/databases/databases.doit.db

*All database tasks up-to-date.*

#### Info
* Doit Database: /home/bjp/transcripts.fasta.dammit/annotate.doit.db
* Input Transcriptome: /home/bjp/transcripts.fasta.dammit/transcripts.fasta

Some tasks out of date!

Up-to-date tasks:
* BUSCO-tetrapoda
* TransDecoder.LongOrfs
* TransDecoder.Predict
* cmscan:Rfam
* gff3:Pfam-A
* gff3:Rfam
* hmmscan:Pfam-A
* hmmscan:Pfam-A:remap
* rename-transcriptome
* transcriptome-stats

Out-of-date tasks:
* annotate:fasta
* gff3:OrthoDB
* gff3:merge-all
* lastal:OrthoDB
* lastal:best-hits:OrthoDB

#### Run Tasks
- [x] transcripts.fasta: 
    * Python: function get_rename_transcriptome_task.fix
- [x] transcriptome_stats:transcripts.fasta: 
    * Python: function get_transcriptome_stats_task.cmd
- [x] busco:transcripts.fasta-tetrapoda_odb9: 
    * Cmd: `BUSCO_CONFIG_FILE="/home/bjp/python_venv_3.5/lib/python3.5/site-packages/dammit/busco_config.ini" python3 /home/bjp/busco/scripts/run_BUSCO.py -i /home/bjp/transcripts.fasta.dammit/transcripts.fasta -f -o transcripts.fasta.tetrapoda.busco.results -l /home/bjp/.dammit/databases/busco2db/tetrapoda_odb9 -m tran -c 24`
- [x] TransDecoder.LongOrfs:transcripts.fasta: 
    * Cmd: `/media/bjp/bioconda/miniconda/bin/TransDecoder.LongOrfs -t /home/bjp/transcripts.fasta.dammit/transcripts.fasta -m 80`
- [x] hmmscan:longest_orfs.pep.x.Pfam-A.hmm: 
    * Cmd: `cat /home/bjp/transcripts.fasta.dammit/transcripts.fasta.transdecoder_dir/longest_orfs.pep | /media/bjp/bioconda/miniconda/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 24 -a /home/bjp/transcripts.fasta.dammit/transcripts.fasta.transdecoder_dir/longest_orfs.pep /media/bjp/bioconda/miniconda/bin/hmmscan --cpu 1 --domtblout /dev/stdout -E 1e-05 -o /home/bjp/transcripts.fasta.dammit/transcripts.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl.hmmscan.out /home/bjp/.dammit/databases/Pfam-A.hmm /dev/stdin > /home/bjp/transcripts.fasta.dammit/transcripts.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl`
- [x] remap_hmmer:longest_orfs.pep.x.pfam.tbl: 
    * Python: function get_remap_hmmer_task.cmd
- [x] hmmscan-gff3:transcripts.fasta.x.pfam-A.gff3: 
    * Cmd: `rm -f /home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.pfam-A.gff3`
    * Python: function get_hmmscan_gff3_task.cmd
- [x] TransDecoder.Predict:transcripts.fasta: 
    * Cmd: `/media/bjp/bioconda/miniconda/bin/TransDecoder.Predict -t /home/bjp/transcripts.fasta.dammit/transcripts.fasta --retain_pfam_hits /home/bjp/transcripts.fasta.dammit/transcripts.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl`
- [x] cmscan:transcripts.fasta.x.Rfam.cm: 
    * Cmd: `cat /home/bjp/transcripts.fasta.dammit/transcripts.fasta | /media/bjp/bioconda/miniconda/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 24 -a /home/bjp/transcripts.fasta.dammit/transcripts.fasta /media/bjp/bioconda/miniconda/bin/cmscan --cpu 1 --rfam --nohmmonly -E 1e-05 --tblout /dev/stdout -o /home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.rfam.tbl.cmscan.out /home/bjp/.dammit/databases/Rfam.cm /dev/stdin > /home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.rfam.tbl`
- [x] cmscan-gff3:transcripts.fasta.x.rfam.gff3: 
    * Cmd: `rm -f /home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.rfam.gff3`
    * Python: function get_cmscan_gff3_task.cmd
- [ ] lastal:/home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.OrthoDB.maf: 
    * Python: function setup_profiler.add_profile_actions.start_profiling
    * Cmd: `cat /home/bjp/transcripts.fasta.dammit/transcripts.fasta | /media/bjp/bioconda/miniconda/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 24 -a /home/bjp/transcripts.fasta.dammit/transcripts.fasta /media/bjp/bioconda/miniconda/bin/lastal -F15 -D100000.0 /home/bjp/.dammit/databases/aa_seq_euk.fasta > /home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.OrthoDB.maf`
    * Python: function setup_profiler.add_profile_actions.stop_profiling
TaskError - taskid:lastal:/home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.OrthoDB.maf
Command error: 'cat /home/bjp/transcripts.fasta.dammit/transcripts.fasta | /media/bjp/bioconda/miniconda/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 24 -a /home/bjp/transcripts.fasta.dammit/transcripts.fasta /media/bjp/bioconda/miniconda/bin/lastal -F15 -D100000.0 /home/bjp/.dammit/databases/aa_seq_euk.fasta > /home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.OrthoDB.maf' returned 141

########################################
TaskError - taskid:lastal:/home/bjp/transcripts.fasta.dammit/transcripts.fasta.x.OrthoDB.maf
<stderr>:
parallel: Warning: A record was longer than 1048576. Increasing to --blocksize 1363150.
parallel: Warning: A record was longer than 1363150. Increasing to --blocksize 1772096.
parallel: Warning: A record was longer than 1772096. Increasing to --blocksize 2303726.
parallel: Warning: A record was longer than 2303726. Increasing to --blocksize 2994845.
parallel: Warning: A record was longer than 2994845. Increasing to --blocksize 3893300.
parallel: Warning: A record was longer than 3893300. Increasing to --blocksize 5061291.
parallel: Warning: A record was longer than 5061291. Increasing to --blocksize 6579680.
parallel: Warning: A record was longer than 6579680. Increasing to --blocksize 8553585.
parallel: Warning: A record was longer than 8553585. Increasing to --blocksize 11119662.
parallel: Warning: A record was longer than 11119662. Increasing to --blocksize 14455562.
parallel: Warning: A record was longer than 14455562. Increasing to --blocksize 18792232.
parallel: Warning: A record was longer than 18792232. Increasing to --blocksize 24429903.
/media/bjp/bioconda/miniconda/bin/lastal: can't read file: /home/bjp/.dammit/databases/aa_seq_euk.fasta.prj
/media/bjp/bioconda/miniconda/bin/lastal: can't read file: /home/bjp/.dammit/databases/aa_seq_euk.fasta.prj
/media/bjp/bioconda/miniconda/bin/lastal: can't read file: /home/bjp/.dammit/databases/aa_seq_euk.fasta.prj
/media/bjp/bioconda/miniconda/bin/lastal: can't read file: /home/bjp/.dammit/databases/aa_seq_euk.fasta.prj
/media/bjp/bioconda/miniconda/bin/lastal: can't read file: /home/bjp/.dammit/databases/aa_seq_euk.fasta.prj
/media/bjp/bioconda/miniconda/bin/lastal: can't read file: /home/bjp/.dammit/databases/aa_seq_euk.fasta.prj

<stdout>:
$

Please advise.

Thanks again!
best,
bjp

**Comment by ljcohen** [comment]
Same issue here. On blank Jetstream instance, Ubuntu 16.04 m1.xlarge (CPU: 24, Mem: 60 GB, Disk: 60 GB). Installed from master.

```
# dammit
## a tool for easy de novo transcriptome annotation

by Camille Scott

**v1.0rc2**, 2018

## submodule: annotate
### Database Check
#### Info
* Database Directory: /home/ljcohen/.dammit/databases
* Doit Database: /home/ljcohen/.dammit/databases/databases.doit.db


*All database tasks up-to-date.*

### Annotation
#### Info
* Doit Database: /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/annotate.doit.db
* Input Transcriptome: /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta


Some tasks out of date!


Up-to-date tasks:
* TransDecoder.LongOrfs
* TransDecoder.Predict
* cmscan:Rfam
* gff3:Pfam-A
* gff3:Rfam
* hmmscan:Pfam-A
* hmmscan:Pfam-A:remap
* rename-transcriptome
* transcriptome-stats


Out-of-date tasks:
* BUSCO-metazoa
* annotate:fasta
* gff3:OrthoDB
* gff3:merge-all
* lastal:OrthoDB
* lastal:best-hits:OrthoDB

#### Run Tasks
- [x] MMETSP0789..trinity_out_2.2.0.Trinity.fasta: 
    * Python: function get_rename_transcriptome_task.fix
- [x] transcriptome_stats:MMETSP0789..trinity_out_2.2.0.Trinity.fasta: 
    * Python: function get_transcriptome_stats_task.cmd
- [ ] busco:MMETSP0789..trinity_out_2.2.0.Trinity.fasta-metazoa_odb9: 
    * Cmd: `python3 /home/ljcohen/busco/scripts/run_BUSCO.py -i /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta -f -o MMETSP0789..trinity_out_2.2.0.Trinity.fasta.metazoa.busco.results -l /home/ljcohen/.dammit/databases/busco2db/metazoa_odb9 -m tran -c 4`
- [x] TransDecoder.LongOrfs:MMETSP0789..trinity_out_2.2.0.Trinity.fasta: 
    * Cmd: `/usr/bin/TransDecoder.LongOrfs -t /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta -m 80`
- [x] hmmscan:longest_orfs.pep.x.Pfam-A.hmm: 
    * Cmd: `cat /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.transdecoder_dir/longest_orfs.pep | /home/ljcohen/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 4 -a /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.transdecoder_dir/longest_orfs.pep /usr/bin/hmmscan --cpu 1 --domtblout /dev/stdout -E 1e-05 -o /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl.hmmscan.out /home/ljcohen/.dammit/databases/Pfam-A.hmm /dev/stdin > /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl`
- [x] remap_hmmer:longest_orfs.pep.x.pfam.tbl: 
    * Python: function get_remap_hmmer_task.cmd
- [x] hmmscan-gff3:MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.pfam-A.gff3: 
    * Cmd: `rm -f /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.pfam-A.gff3`
    * Python: function get_hmmscan_gff3_task.cmd
- [x] TransDecoder.Predict:MMETSP0789..trinity_out_2.2.0.Trinity.fasta: 
    * Cmd: `/usr/bin/TransDecoder.Predict -t /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta --retain_pfam_hits /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.transdecoder_dir/longest_orfs.pep.x.pfam.tbl`
- [x] cmscan:MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.Rfam.cm: 
    * Cmd: `cat /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta | /home/ljcohen/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 4 -a /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta /usr/bin/cmscan --cpu 1 --rfam --nohmmonly -E 1e-05 --tblout /dev/stdout -o /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.rfam.tbl.cmscan.out /home/ljcohen/.dammit/databases/Rfam.cm /dev/stdin > /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.rfam.tbl`
- [x] cmscan-gff3:MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.rfam.gff3: 
    * Cmd: `rm -f /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.rfam.gff3`
    * Python: function get_cmscan_gff3_task.cmd
- [ ] lastal:/home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.OrthoDB.maf: 
    * Python: function setup_profiler.add_profile_actions.start_profiling
    * Cmd: `cat /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta | /home/ljcohen/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 4 -a /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta /home/ljcohen/last-658/src/lastal -F15 -D100000.0 /home/ljcohen/.dammit/databases/aa_seq_euk.fasta > /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.OrthoDB.maf`
    * Python: function setup_profiler.add_profile_actions.stop_profiling
TaskError - taskid:lastal:/home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.OrthoDB.maf
Command error: 'cat /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta | /home/ljcohen/bin/parallel --round-robin --pipe -L 2 -N 10000 --gnu -j 4 -a /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta /home/ljcohen/last-658/src/lastal -F15 -D100000.0 /home/ljcohen/.dammit/databases/aa_seq_euk.fasta > /home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.OrthoDB.maf' returned 141

########################################
TaskError - taskid:lastal:/home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.OrthoDB.maf
lastal:/home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.OrthoDB.maf <stderr>:
parallel: Warning: A record was longer than 1048576. Increasing to --blocksize 1363150.
parallel: Warning: A record was longer than 1363150. Increasing to --blocksize 1772096.
parallel: Warning: A record was longer than 1772096. Increasing to --blocksize 2303726.
parallel: Warning: A record was longer than 2303726. Increasing to --blocksize 2994845.
parallel: Warning: A record was longer than 2994845. Increasing to --blocksize 3893300.
parallel: Warning: A record was longer than 3893300. Increasing to --blocksize 5061291.
parallel: Warning: A record was longer than 5061291. Increasing to --blocksize 6579680.
parallel: Warning: A record was longer than 6579680. Increasing to --blocksize 8553585.
parallel: Warning: A record was longer than 8553585. Increasing to --blocksize 11119662.

lastal:/home/ljcohen/mmetsp_assemblies_March2018/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.dammit/MMETSP0789..trinity_out_2.2.0.Trinity.fasta.x.OrthoDB.maf <stdout>:
```

**Comment by desmondramirez** [comment]
Hi @DrPintoThe2nd, @ljcohen, @camillescott 

I've gotten this same error as the other posters with lastal. Has anyone found a workaround/solved this issue?

Thanks,
Des

**Comment by DrPintoThe2nd** [comment]
No, sorry. I gave up a long time ago!
Best,
bjp

**Comment by desmondramirez** [comment]
Hi @DrPintoThe2nd @ljcohen @camillescott, 

I figured out why the lastal command failed- for whatever reason, when installing the databases, lastdb was running out of memory and never finished, so the files lastal was trying to call didn't work properly. The workaround was to run lastdb with the `-P` flag to give 12 threads so that it could finish. Once I did that, the dammit pipeline completed! A way to avoid this when installing the databases may be to run the `dammit databases --install' with more threads if possible. I'm working on a cluster, so if I were to do it again, I'd install the dammit databases in a cluster job.

Here's the command that finally worked to finish installing the lastdb databases: `dammit/bin/lastdb -p -w3 -P 12 -v ./aa_seq_euk.fasta ./aa_seq_euk.fasta`

Hope this helps!
